### PR TITLE
TinyGo HTTP Server Actor Abstraction

### PR DESCRIPTION
**Pull request by jordan-rash**
Included a test script that I have successfully been using locally.

### CODE REVIEW

File: http-server/go/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# wasmCloud Core Actor Interface
```
Comment (autodidaddict) [inline http-server/go/README.md:1]: This should be `HTTP Server` rather than core.

File: http-server/go/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+This actor interface provides wasmCloud actors with an interface to the HTTP Server capability provider. Actors using this
+interface must have the claim `wasmcloud:httpserver` in order to have permission to handle requests, and they
+must have an active, configured binding to an HTTP Server capability provider.
```
Comment (autodidaddict) [inline http-server/go/README.md:5]: s/binding/link. I know the old README you copied this from says binding, but that's "old" (pre-0.15) terminology.

File: http-server/go/httpserver.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	binding string
+}
+
+func NewHost(binding string) *Host {
+	return &Host{
+		binding: binding,
+	}
+}
+
+func (h *Host) HandleRequest(request Request) (Response, error) {
```
Comment (autodidaddict) [inline http-server/go/httpserver.go:18]: we should remove this, since guests can't call `HandleRequest` on the host.

**Review by autodidaddict — CHANGES_REQUESTED** [review]

**Review by autodidaddict — APPROVED** [review]

**Review by autodidaddict — APPROVED** [review]

**Comment by jordan-rash** [comment]
This should satisfy #5

**Pull request merged**